sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracerManagement.java

```diff
@@ -64,8 +64,11 @@
    * <p>After this is called, further attempts at re-using or reconfiguring this instance will
    * result in undefined behavior. It should be considered a terminal operation for the SDK
    * implementation.
+   *
+   * @return a {@link CompletableResultCode} which is completed when all the span processors have
+   *     been shut down.
    */
-  void shutdown();
+  CompletableResultCode shutdown();
```

Comment:
Does this make impossible to change to `close` and implement `AutoClosable`? Do we care about that?

Reply:
It shouldn't. The goal here was to make it so that we could have 2 modes of "closing". One is "shutdown()" like we have today, which will universally return `CompletableResultCode`, and then the `close` method proposed in #2285 as the auto-closable version.